### DISCUSSION TITLE
[Closed] Add `.PlaceHolder()` Extension Methods

### DISCUSSION THREAD

**Discussion by TheCodeTraveler** [CommunityToolkit/Maui.Markup#41] [New Feature Discussions]
After adding #38, it makes sense to also implement extension methods for `Placeholder` and `PlaceholderColor`:

```cs
	public static TBindable PlaceholderColor<TBindable>(this TBindable bindable, Color? textColor) where TBindable : BindableObject, IPlaceholder
	{
		bindable.SetValue(PlaceholderElement.PlaceholderColorProperty, textColor);
		return bindable;
	}

	public static TBindable Placeholder<TBindable>(this TBindable bindable, string? text) where TBindable : BindableObject, IPlaceholder
	{
		bindable.SetValue(PlaceholderElement.PlaceholderProperty, text);
		return bindable;
	}

	public static TBindable Placeholder<TBindable>(this TBindable bindable, string? text, Color? textColor) where TBindable : BindableObject, IPlaceholder
	{
		return bindable.Placeholder(text).PlaceholderColor(textColor);
	}
```

**Comment by TheCodeTraveler** [comment]
A preview of the implementation and associated Unit Tests can be found on the [`Create-PlaceholderExtensions` branch](https://github.com/CommunityToolkit/Maui.Markup/tree/Create-PlaceholderExtensions).

**Comment by pictos** [comment]
`public static TBindable Placeholder<TBindable>(this TBindable bindable, string? text);`

This overload shouldn't be named `PlaceholderText`? I'm asking because we have a specific name of `Color`

**Reply by TheCodeTraveler** [reply]
I went with the method name `.Placeholder()` because the `IPlaceholder` property name is `Placeholder`:

```cs
new Entry
{
  Placeholder = "Enter Text Here"
};
```

This matches the existing precedence, like `.FormattedText()` sets the `Label.FormattedText` property, and `.FontSize()` sets the `Label.FontSize` property.

Since the `.Placeholder()` extension method will only appear for elements that implement `IPlaceholder`, it should always make sense to the user given the context:

```cs
new Entry().Placeholder("Name");
new Editor().Placeholder("Address");
```

**Reply by TheCodeTraveler** [reply]
Likewise, with `.PlaceholderColor()`; it matches the `IPlaceholder` property name, `IPlaceholder.PlaceholderColor`

**Reply by pictos** [reply]
Yeah, that makes sense to me.

**Reply by TheCodeTraveler** [reply]
Right on! Is it good enough for a Proposal?

**Reply by pictos** [reply]
Yeap! This has my approval!

**Comment by TheCodeTraveler** [comment]
Promoted to Proposal: https://github.com/CommunityToolkit/Maui.Markup/issues/43

**Comment by CliffAgius** [comment]
Closed and answered
